client/components/payment-process/payment-process.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable react/require-default-props */
+import {Cookies} from "react-cookie";
+import PropTypes from "prop-types";
+import React from "react";
+import {Redirect} from "react-router-dom";
+import {toast} from "react-toastify";
+import {t} from "ttag";
+import LoadingContext from "../../utils/loading-context";
+import validateToken from "../../utils/validate-token";
+import getPaymentStatusRedirectUrl from "../../utils/get-payment-status";
+import history from "../../utils/history";
+
+export default class PaymentProcess extends React.Component {
+  constructor(props) {
+    super(props);
+    this.iframeRef = React.createRef();
+    this.state = {
+      isTokenValid: null,
+      iframeHeight: 450,
+    };
+  }
+
+  async componentDidMount() {
+    const {cookies, orgSlug, setUserData, logout} = this.props;
+    let {userData} = this.props;
+    const {setLoading} = this.context;
+
+    setLoading(true);
+    toast.info(t`PLEASE_WAIT`, {autoClose: 10000});
+    const isTokenValid = await validateToken(
+      cookies,
+      orgSlug,
+      setUserData,
+      userData,
+      logout,
+    );
+    this.setState({isTokenValid});
+    if (isTokenValid === false) {
+      return;
+    }
+
+    ({userData} = this.props);
+    setUserData({...userData});
+    window.addEventListener("message", this.handlePostMessage);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("message", this.handlePostMessage);
+  }
+
+  handlePostMessage = async (event) => {
+    const {userData, cookies, orgSlug, setUserData} = this.props;
+    const {setLoading} = this.context;
+    const {message, type} = event.data;
+    // For security reasons, read https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#security_concern
+    if (
+      (userData.payment_url &&
+        event.origin === new URL(userData.payment_url).origin) ||
+      event.origin === window.location.origin
+    ) {
+      if (type === "paymentClose") {
```

Comment:
I think we can now change to a switch statement since the possible cases are 3 now (I added setHeight)